File: المسار العاشر __ إعادة تصميم البرمجيات/RefactoringTech/Custom Exception/custom_exception.js
const url = "https://app.reqres.in/api/users/2000";

class UserNotFoundException extends Error{
    constructor(message) {
        super(message);
        this.name = 404;
    }
}

fetch(url)
  .then((response) => {
    if (response.status == 404) {
      throw new UserNotFoundException("User not Found");
    }
    return response.json();
  })
  .then((userData) => {
    console.log("User Data:", userData);
  })
  .catch((error) => {
    console.log("Error:", error);
  });

File: المسار العاشر __ إعادة تصميم البرمجيات/RefactoringTech/Custom Exception/custom_exception.py
import requests

class UserNotFoundException(Exception):
    def __init__(self, message='404: User Not Found!'):
        self.message = message
        super().__init__(self.message)



def get_user(id):
    url = f"https://app.reqres.in/api/users/{id}"
    response = requests.get(url)
    if response.status_code == 404:
        raise UserNotFoundException()
    return response.json()


print(get_user(20))